website/docs/metaprogramming-plugins.md

```diff
@@ -0,0 +1,124 @@
+---
+id: metaprogramming-plugins
+title: Metaprogramming plugins
+---
+
+> **Note**: The plugin system has many drawbacks and is not considered a stable
+> feature. Expect its API to change in the future.
+
+In Ruby, it is possible to dynamically define methods. Consider the following
+example:
+
+```ruby
+# metaprogramming.rb
+# typed: true
+
+class MetaProgramming
+  def self.macro(name)
+    define_method("#{name}_time") { name }
+  end
+
+  macro(:bed)
+  macro(:fun)
+end
+
+hello = MetaProgramming.new
+hello.bed_time # error: Method `bed_time` does not exist on MetaProgramming
+hello.fun_time # error: Method `fun_time` does not exist on MetaProgramming
+```
+
+Sorbet rejects this script since currently, Sorbet does not try to understand
```

Comment:
```suggestion
Sorbet rejects this script since Sorbet cannot understand
```

It's unlikely that this will ever change, and all docs are assumed to document what the behavior is "currently".